arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/raise/Builders.kt

```diff
@@ -47,12 +50,12 @@ public inline fun <Error, A> either(@BuilderInference block: Raise<Error>.() ->
  * Read more about running a [Raise] computation in the
  * [Arrow docs](https://arrow-kt.io/learn/typed-errors/working-with-typed-errors/#running-and-inspecting-results).
  *
- * @see NullableRaise.ignoreErrors By default, `nullable` only allows raising `null`.
- * Calling [ignoreErrors][NullableRaise.ignoreErrors] inside `nullable` allows to raise any error, which will be returned to the caller as if `null` was raised.
+ * @see SingletonRaise.ignoreErrors By default, `nullable` only allows raising `null`.
+ * Calling [ignoreErrors][SingletonRaise.ignoreErrors] inside `nullable` allows to raise any error, which will be returned to the caller as if `null` was raised.
  */
-public inline fun <A> nullable(block: NullableRaise.() -> A): A? {
+public inline fun <A> nullable(block: SingletonRaise<Nothing?>.() -> A): A? {
```

Comment:
Should we be allowing `raise(null)`? I struggle to see any uses for this, perhaps outside of someone having an `Either<Nothing?, R>`, but that seems rare. Should this instead be `SingletonRaise<Unit>`? I much prefer using Unit to denote that you shouldn't be raising any real values. This would also help the transition into context receivers eventually because all the APIs on `SingletonRaise` would be applicable to `Raise<Unit>`

Reply:
Keeping this one is important, because we really want the `Raise` parameter to reflect the error type, which is `None` in the case of `Option`, for example. Breaking this would also break users of Arrow 1.x with no apparent reason.

Reply:
That's fair. I'm questioning whether users of Arrow 1.x used `raise(None)` and `raise(null)`. Maybe we can add extra methods for compatibility?

Reply:
I'm not sure, but we would still be breaking compatibility. We promised that code compiled with 1.2.x would be source-compatible with 2.0 (except for deprecated methods), so we should keep (1) the `raise` with argument, (2) the fact that they are `Raise<Nothing?>` and `Raise<None>`

Reply:
Good point. `Raise<Nothing?>` and `Raise<None>` would likely show up in people's code if they were dealing with `nullable` and `option`. I think in a context world, we can have `nullable` introduce `context(Raise<Unit>, Raise<Nothing?>)` and same for `option`